ID-B-4.4

AlexeyVolkov2023 opened the pull request:
Создал классы Breed, Spacies, создал рекорды Breedid, Speciesid, IdentifierAnimal. Написал конфигурацию для Breed и Species, поправил конфигурацию Animal

AlexeyVolkov2023 committed:
ID-B-4.4

Создал классы Breed, Spacies, создал рекорды Breedid, Speciesid, IdentifierAnimal. Написал конфигурацию для Breed и Species, поправил конфигурацию Animal

AlexeyVolkov2023 commented on backend/src/HelpAnimal.Domain/Models/IdentifierAnimal.cs:
```diff
@@ -0,0 +1,25 @@
+    public IdentifierAnimal()
+    {
+    }
+
```
Получилось сделать миграцию только после добавления пустого конструктора, но так как это VO EF же не должен создавать отдельную таблицу

YADARAN1 replied:
У тебя возможно ошибка из-за несоответствия имён, у тебя в конструкторах(я в breed увидел Breedid), измени везде на обычное название "id"

YADARAN1 replied:
И потом протестируй, не забудь, удалить конструктор публичный из IdentifierAnimal и второй сделай приватным

YADARAN1 commented on backend/src/HelpAnimal.Infrastructure/Configuration/BreedConfiguration.cs:
```diff
@@ -0,0 +1,25 @@
+{
+    public void Configure(EntityTypeBuilder<Breed> builder)
+    {
+        builder.ToTable("Breeds");
```
snakecase, всё с маленькой буквы

YADARAN1 commented on backend/src/HelpAnimal.Infrastructure/Configuration/SpeciesConfiguration.cs:
```diff
@@ -0,0 +1,27 @@
+{
+    public void Configure(EntityTypeBuilder<Species> builder)
+    {
+        builder.ToTable("Species");
```
такая же история, всё с маленькой

YADARAN1 commented on backend/src/HelpAnimal.Infrastructure/HelpAnimalDbContext.cs:
```diff
@@ -7,29 +7,28 @@
-  
-    
-    
+    public DbSet<Species> Specieses { get; set; }
```
убери DbSet<Animal> Animals, мы через волонтёра(агрегат) взаимодействуем с животным

AlexeyVolkov2023 committed:
ID-B-4.4

исправил имена классов, сделаал исправления насчет speciesid breedid  но миграция все равно требует пустой конструктор , подозреваю что дело в конфигурации , но не могу понять в чем

AlexeyVolkov2023 commented on backend/src/HelpAnimal.Infrastructure/Migrations/HelpAnimalDbContextModelSnapshot.cs:
```diff
@@ -76,52 +76,6 @@ protected override void BuildModel(ModelBuilder modelBuilder)
-                                        .HasColumnName("identifier_species_id_value");
-                                });
-                        });
-
```
Пока делал конфигурацию то создал две миграции без аптдейта, потом даже с удалением миграций не смог ничего сделать . пробовал удалить вольюм из докера , но образ почему-то не видел новый файл, хотя имя было такое-же . В конце-концов удалили и запустил образ заново

YADARAN1 replied:
Скинь в технические вопросы, поможем тебе там

YADARAN1 commented on backend/src/HelpAnimal.Domain/Models/AnimalPhotosDetails.cs:
```diff
@@ -1,7 +1,7 @@
+public class AnimalPhotosDetails
 {
-   public List<AnimalPhoto> Photos { get; }  
+   public List<AnimalPhoto>? Photos { get; } 
```
Вот тут лучше писать не List, a IReadOnlyList, и убери знак вопроса

YADARAN1 commented on backend/src/HelpAnimal.Domain/Models/PhoneNumber.cs:
```diff
@@ -5,26 +5,29 @@ namespace HelpAnimal.Domain.Models;
         if (!Regex.IsMatch(number, PHONE_NUMBER_REGEX))
         {
-            throw new ArgumentException("Phone number is invalid. It should be 10-15 digits long, possibly with a '+' prefix.", nameof(number));
+            throw new ArgumentException(
```
У тебя не должен здесь выкидываться Exception, добавь регулярное выражение

YADARAN1 commented on backend/src/HelpAnimal.Domain/Models/RequisiteDetails.cs:
```diff
@@ -2,5 +2,5 @@
 public record RequisiteDetails
 {
-    public List<Requisite> Requisites { get; private set; }
+    public List<Requisite>? Requisites { get; private set; }
```
Тут тоже поменяй на IReadOnlyList и без знака вопроса + убери private set

YADARAN1 commented on backend/src/HelpAnimal.Domain/Models/SocialDetails.cs:
```diff
@@ -2,5 +2,5 @@
 public record SocialDetails
 {
-    public List<SocialNetwork> Networks { get; private set;}
+    public List<SocialNetwork>? Networks { get; private set;}
```
Тут тоже поменяй на IReadOnlyList и без знака вопроса + убери private set

YADARAN1 commented on backend/src/HelpAnimal.Domain/Models/Species.cs:
```diff
@@ -2,30 +2,30 @@
     }
 
-    public string Title { get;  } 
+    public string? Title { get;  } 
```
знак вопроса убрать

YADARAN1 commented on backend/src/HelpAnimal.Domain/Models/VaccinationDetails.cs:
```diff
@@ -2,5 +2,5 @@
 public class VaccinationDetails
 {
-    public List<Vaccination> Vaccinations { get;  private set; }
+    public List<Vaccination>? Vaccinations { get;  }
```
Тут тоже поменяй на IReadOnlyList и без знака вопроса

YADARAN1 commented on backend/src/HelpAnimal.Domain/Models/Volunter.cs:
```diff
@@ -35,17 +35,17 @@ private Volunteer(
+    public int AdoptedAnimalsCount { get; private set; }
+    public int CurrentAnimalsCount { get; private set; }
+    public int AnimalsInTreatmentCount { get; private set; }
+    public PhoneNumber? Phone { get; private set; }
```
Убрать знак вопроса

YADARAN1 replied:
Тут тоже поменяй на IReadOnlyList и без знака вопроса и добавь private readonly List<Pet> _pets = []

YADARAN1 commented on backend/src/HelpAnimal.Domain/Models/Volunter.cs:
```diff
@@ -35,17 +35,17 @@ private Volunteer(
     public SocialDetails? SocialNetworks { get; private set; }
     public RequisiteDetails? RequisiteCollection { get; private set; }
-    public List<Animal>? Animals { get; private set; } = []; // Список домашних животных
+    public List<Animal>? Animals { get; private set; }
```
Убрать знак вопроса

YADARAN1 commented on backend/src/HelpAnimal.Domain/Models/Volunter.cs:
```diff
@@ -83,7 +83,7 @@ public static Result<Volunteer> Create(
 
         return new Volunteer(
-            volunteerid,
+            id,
```
![image](https://github.com/user-attachments/assets/a0c431de-ded2-4a22-9a62-98641aa6f174)
Нам при создании волонтёра не нужны животные, убери их

YADARAN1 commented on backend/src/HelpAnimal.Infrastructure/Configuration/AnimalConfiguration.cs:
```diff
@@ -26,11 +26,15 @@ public void Configure(EntityTypeBuilder<Animal> builder)
         {
-            ai.ComplexProperty(b => b.SpeciesId, c => { c.Property(d => d.Value); });
-            ai.Property(d => d.BreedId);
+            ai.ToJson();
```
в скобках название в snakecase стиле

YADARAN1 commented on backend/src/HelpAnimal.Infrastructure/Configuration/AnimalConfiguration.cs:
```diff
@@ -40,14 +44,14 @@ public void Configure(EntityTypeBuilder<Animal> builder)
 
 
-        builder.ComplexProperty(a => a.AnimalAddress, aa =>
+        builder.OwnsOne(a => a.AnimalAddress, aa =>
```
Сконфигурируй как ComplexProperty

YADARAN1 commented on backend/src/HelpAnimal.Domain/Models/Animal.cs:
```diff
@@ -47,24 +47,24 @@ private Animal(
     public string Name { get; private set; } = default!;
     public string Description { get; private set; } = default!;
-    public IdentifierAnimal Identifier { get; private set; }
+    public IdentifierAnimal? Identifier { get; private set; }
```
Убрать знак вопроса

YADARAN1 commented on backend/src/HelpAnimal.Domain/Models/Animal.cs:
```diff
@@ -47,24 +47,24 @@ private Animal(
     public string Color { get; private set; } = default!;
     public string HealthInfo { get; private set; } = default!;
-    public Address AnimalAddress { get; private set; } = default!;
+    public Address? AnimalAddress { get; private set; } = default!;
```
Убрать знак вопроса

YADARAN1 commented on backend/src/HelpAnimal.Domain/Models/Animal.cs:
```diff
@@ -47,24 +47,24 @@ private Animal(
     public double Weight { get; private set; }
     public double Height { get; private set; }
-    public PhoneNumber Phone { get; private set; } // Номер телефона для связи с владельцем
+    public PhoneNumber? Phone { get; private set; } // Номер телефона для связи с владельцем
```
Убрать знак вопроса

YADARAN1 commented on backend/src/HelpAnimal.Domain/Models/Animal.cs:
```diff
@@ -47,24 +47,24 @@ private Animal(
     public bool IsNeutered { get; private set; } // Кастрирован или нет
     public DateTime DateOfBirth { get; private set; }
-    public VaccinationDetails AlreadyVaccinated { get; private set; }
+    public VaccinationDetails? AlreadyVaccinated { get; private set; }
```
Убрать знак вопроса

KirillSachkov reviewed (changes requested):
Сделать исправления

AlexeyVolkov2023 committed:
Comments

работа с замечаниями
